multiqc/modules/picard/ValidateSamFile.py

```diff
@@ -0,0 +1,369 @@
+            only_warning_count += 1
+        else:
+            pass_count += 1
+
+    # Add overview note
+    plot_html = []
+    note_html = _generate_overview_note(
+                        pass_count=pass_count,
+                        only_warning_count=only_warning_count,
+                        error_count=error_count,
+                        total_count=len(data)
+                    )
+    plot_html.append(note_html)
+
+    # Add the detailed table, but only if we have something to show
+    if error_count or only_warning_count:
+        table_html = _generate_detailed_table(data)
+        plot_html.append(table_html)
+
+    # Finally, add the html to the report as a section
+    self.add_section(
+            name='SAM/BAM File Validation',
+            anchor='picard_validatesamfile',
+            description='This tool reports on the validity of a SAM or BAM file relative to the SAM-format specification.',
+            helptext='''
+            A detailed table is only shown if a errors or warnings are found. Details about the errors and warnings are only shown if a SUMMARY report was parsed.
+
+            For more information on the warnings, errors and possible fixes please read [this broadinstitute article](https://software.broadinstitute.org/gatk/documentation/article.php?id=7571).''',
+            plot="\n".join(plot_html),
+    )
+
+
+def _add_data_to_general_stats(self, data):
+    """
+        Add data for the general stats in a Picard-module specific manner
+    """
+    headers = _get_general_stats_headers()
+
+    self.general_stats_headers.update(headers)
+
+    general_data = dict()
+    for sample in data:
+        general_data[sample] = {key: data[sample][key] for key in ('ERROR_count', 'WARNING_count', 'file_validation_status')}
+
+        if sample not in self.general_stats_data:
+                self.general_stats_data[sample] = dict()
+        self.general_stats_data[sample].update(general_data[sample])
+
+    self.general_stats_addcols(data=general_data, headers=headers)
+
+
+def _get_general_stats_headers():
+    """
+        Returns a header dict for the general stats collumns
+    """
+    headers = OrderedDict()
+
+    headers['file_validation_status'] = {
+        'title': "File-Validation Status",
+        'description': 'pass/warn/fail',
+        'hidden': False,
+    }
+
+    headers['WARNING_count'] = {
+        'title': "# Validation Warnings",
+        'description': 'number of warnings',
+        'scale': 'Set3',
+        'max': 1,
+        'colour': '255,237,160',
+        'format': '{:.0f}',
+        'hidden': True,
+    }
+
+    headers['ERROR_count'] = {
+        'title': "# Validation Errors",
+        'description': 'number of errors',
+        'scale': 'RdYlGn-rev',
+        'max': 1,
+        'colour': '252,146,114',
+        'format': '{:.0f}',
+        'hidden': True,
+    }
+
+    return headers
+
+
+def _generate_overview_note(pass_count, only_warning_count, error_count, total_count):
```

Comment:
Nice approach with the custom HTML! It might be nicer to have a stacked progress bar to show the proportions... What do you think? eg. https://getbootstrap.com/docs/3.3/components/#progress-stacked

Reply:
I think this is indeed nicer. It comes with the problem of wanting to label them. I changed it to use a progress bar: ad978a04